api/desecapi/exceptions.py

```diff
@@ -1,17 +1,28 @@
 import json
 from json import JSONDecodeError
 
-from rest_framework.exceptions import APIException
+from rest_framework import status
+from rest_framework.exceptions import APIException, ValidationError
 
 
-class PDNSException(APIException):
+class RequestEntityTooLarge(APIException):
+    status_code = status.HTTP_413_REQUEST_ENTITY_TOO_LARGE
+    default_detail = 'Payload too large.'
+    default_code = 'too_large'
+
+
+class PDNSValidationError(ValidationError):
+    status_code = 422
+
+    def __init__(self, response=None):
+        try:
+            detail = json.loads(response.text)['error']
+        except JSONDecodeError:
```

Comment:
add KeyError

Reply:
I did not add KeyError because we expect it to always be present. If it is not, we should have this turn into a 500 error so that we can investigate what's going on. Ok for you?

Reply:
alright, fair point